### ISSUE TITLE
Integer type generated for XSD duration types

### ISSUE THREAD

**Issue by czukowski** [WsdlToPhp/PackageGenerator#65]
Example:

``` xml
<xs:attribute name="Duration" type="xs:duration" use="required">
```

Generated code:

``` php
/**
 * Set Duration value
 * @param int $duration
 * @return $this
 */
public function setDuration($duration = NULL)
{
    // validation for constraint: int
    if (!is_NULL($duration) && !is_numeric($duration)) {
        throw new \InvalidArgumentException(sprintf('Invalid value, please provide a numeric value, "%s" given', gettype($duration)), __LINE__);
    }
    $this->Duration = $duration;
    return $this;
}
```

According to [XSD docs](https://www.w3.org/TR/xmlschema11-2/#duration), durations should be strings like P30D or P5Y or P5Y2M10D.

**Comment by mikaelcom** [comment]
@czukowski Would it be possible for you test this branch so I can merge this fix? Thx

**Comment by czukowski** [comment]
@mikaelcom Thank you for the update, I'll give it a try and let you know.

**Comment by czukowski** [comment]
Looks good to me:

```php
/**
 * Set Duration value
 * @param string $duration
 * @return $this
 */
public function setDuration($duration = null)
{
	// validation for constraint: string
	if (!is_null($duration) && !is_string($duration)) {
		throw new \InvalidArgumentException(sprintf('Invalid value, please provide a string, "%s" given', gettype($duration)), __LINE__);
	}
	$this->Duration = $duration;
	return $this;
}
```

**Comment by mikaelcom** [comment]
closed with tag [1.8.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/1.8.0) and [2.3.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/2.3.0)
